src/trajectory.jl

```diff
@@ -33,9 +33,45 @@ Abstract Markov chain Monte Carlo proposal.
 abstract type AbstractProposal end
 
 """
-Hamiltonian dynamics numerical simulation trajectories.
+Abstract termination criteria.
+"""
+abstract type AbstractTerminationCriterion end
+
+abstract type StaticTerminationCriterion <: AbstractTerminationCriterion end
+
+abstract type DynamicTerminationCriterion <: AbstractTerminationCriterion end
+
+"""
+$(TYPEDEF)
+
+Static HMC with a fixed number of leapfrog steps.
+
+# Fields
+
+$(TYPEDFIELDS)
+
+# References
+1. Neal, R. M. (2011). MCMC using Hamiltonian dynamics. Handbook of Markov chain Monte Carlo, 2(11), 2. ([arXiv](https://arxiv.org/pdf/1206.1901))
 """
-abstract type AbstractTrajectory{I<:AbstractIntegrator} <: AbstractProposal end
+struct FixedNSteps{T<:Integer} <: StaticTerminationCriterion
+    L::T
+end
+
+"""
+$(TYPEDEF)
+
+Standard HMC implementation with fixed total trajectory length.
+
+# Fields
+
+$(TYPEDFIELDS)
+
+# References
+1. Neal, R. M. (2011). MCMC using Hamiltonian dynamics. Handbook of Markov chain Monte Carlo, 2(11), 2. ([arXiv](https://arxiv.org/pdf/1206.1901)) 
+"""
+struct FixedLength{F<:AbstractFloat} <: StaticTerminationCriterion
```

Comment:
I think `FixedIntegrationTime` is a more accurate term here. Maybe rename it?

Reply:
Good suggestion!